File: frontend/src/types/email.ts
export type EmailDirection = "outbound" | "inbound";
export type EmailKind = "verification" | "password_reset" | "admin" | "inbound" | "other";
export type EmailStatus = "sent" | "failed" | "received";

export interface EmailLogEntry {
  id: string;
  direction: EmailDirection;
  kind: EmailKind;
  status: EmailStatus;
  address_from: string;
  address_to: string;
  subject: string;
  body_preview: string;
  user_id: string | null;
  sent_by_id: string | null;
  error: string | null;
  created_at: string;
}

export interface EmailLogPage {
  entries: EmailLogEntry[];
  total: number;
  page: number;
  page_size: number;
}

export interface EmailStats {
  sent_24h: number;
  failed_24h: number;
  sent_30d: number;
  failed_30d: number;
  received_30d: number;
  by_kind: Record<string, number>;
  last_sent_at: string | null;
}

export interface AdminEmailSendResult {
  sent: number;
  failed: number;
  skipped: string[];
}

File: backend/app/schemas/email.py
import uuid

from pydantic import BaseModel, ConfigDict, EmailStr, Field

from app.schemas.common import UTCDateTime


class EmailLogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    direction: str
    kind: str
    status: str
    address_from: str
    address_to: str
    subject: str
    body_preview: str
    user_id: uuid.UUID | None
    sent_by_id: uuid.UUID | None
    error: str | None
    created_at: UTCDateTime


class EmailLogPageOut(BaseModel):
    entries: list[EmailLogOut]
    total: int
    page: int
    page_size: int


class EmailStatsOut(BaseModel):
    sent_24h: int
    failed_24h: int
    sent_30d: int
    failed_30d: int
    received_30d: int
    # Разбивка отправленных за 30 дней по типам писем (verification/password_reset/admin).
    by_kind: dict[str, int]
    last_sent_at: UTCDateTime | None


class AdminEmailSend(BaseModel):
    """Ручное письмо из админки. Получатели задаются пользователями сайта и/или
    произвольными адресами; каждому уходит отдельное письмо, чтобы адреса
    получателей не видели друг друга."""

    user_ids: list[uuid.UUID] = Field(default_factory=list)
    emails: list[EmailStr] = Field(default_factory=list)
    subject: str = Field(min_length=1, max_length=200)
    body: str = Field(min_length=1, max_length=5000)


class AdminEmailSendResult(BaseModel):
    sent: int
    failed: int
    # Пользователи без почты или с неподтверждённым адресом - им не отправляли.
    skipped: list[str]
